hbase-client/src/main/java/org/apache/hadoop/hbase/util/PoolMap.java

```diff
@@ -321,62 +345,44 @@ public R get() {
    * @param <R>
    *          the type of the resource
    */
-  static class ThreadLocalPool<R> extends ThreadLocal<R> implements Pool<R> {
-    private static final Map<ThreadLocalPool<?>, AtomicInteger> poolSizes = new HashMap<>();
+  static class ThreadLocalPool<R> implements Pool<R> {
```

Comment:
Should we rename this class entirely? `ThreadLocal` in the name may mislead developers into thinking this fully behaves like a `ThreadLocal`.

Reply:
I'm hesitant to have us change this name since it appears in configuration. What behavior of ThreadLocal are you concerned about us deviating from?

Reply:
I wouldn't rename it, because the configuration value called 'thread-local'. ThreadLocal removes mapping automatically when thread exits, that is a difference.

Reply:
Ok.